dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java

```diff
@@ -336,32 +338,50 @@ private void clear() {
      */
     private void parseProcessOutput(Process process) {
         String threadLoggerInfoName = String.format(LoggerUtils.TASK_LOGGER_THREAD_NAME + "-%s", taskExecutionContext.getTaskAppId());
-        ExecutorService parseProcessOutputExecutorService = ThreadUtils.newDaemonSingleThreadExecutor(threadLoggerInfoName);
-        parseProcessOutputExecutorService.submit(new Runnable() {
+        ExecutorService getOutputLogService = ThreadUtils.newDaemonSingleThreadExecutor(threadLoggerInfoName + "-" + "getOutputLogService");
+        getOutputLogService.submit(new Runnable() {
             @Override
             public void run() {
                 BufferedReader inReader = null;
 
                 try {
                     inReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
```

Comment:
Is there a problem here? Does ErrorStream cause blocking?

Reply:
I found DS code  has set processBuilder.redirectErrorStream(true);  before.